include/arch/arm/arch.h

```diff
@@ -107,10 +153,21 @@ extern "C" {
  * 2) Used to determine the alignment of a stack buffer
  *
  */
+#define STACK_ALIGN MAX(THREAD_MIN_STACK_ALIGN, MPU_GUARD_ALIGN)
+
+/**
+ * @brief Define alignment of a privilege stack buffer
+ *
+ * This is used to determine the required alignment of threads'
+ * privilege stacks when building with support for user mode.
+ *
+ * @note
+ * The privilege stacks do not need to respect the minimum MPU
+ * region alignment requirement (unless this is enforced via
+ * the MPU Stack Guard feature).
+ */
 #if defined(CONFIG_USERSPACE)
-#define STACK_ALIGN    CONFIG_ARM_MPU_REGION_MIN_ALIGN_AND_SIZE
-#else
-#define STACK_ALIGN    MAX(STACK_ALIGN_SIZE, MPU_GUARD_ALIGN_AND_SIZE)
+#define PRIVILEGE_STACK_ALIGN MAX(STACK_ALIGN_SIZE, MPU_GUARD_ALIGN)
```

Comment:
can you Z_ prefix this and other macros you are introducing in this file so we aren't polluting the namespace

Reply:
Yes, will do, thanks.

Reply:
Fixed now.